[CLN] payment_razorpay_oauth: merge into `payment_razorpay`

nikj-odoo opened the pull request:
This commit merges the Razorpay OAuth flow into the main `payment_razorpay` module, integrating both flows into a single one, and removing the `payment_razorpay_oauth` module.

See also:
- https://github.com/odoo/upgrade/pull/6858

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/188774.png)](https://mergebot.odoo.com/odoo/odoo/pull/188774)

kcv-odoo commented on addons/payment_razorpay/const.py:
```diff
@@ -1,5 +1,8 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+# URL for OAUTH flow
```
No need for this comment as variable name is self explanatory

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -25,18 +31,28 @@ class PaymentProvider(models.Model):
-        required_if_provider='razorpay',
+        required_if_provider=False,
         groups='base.group_system',
     )
```
```suggestion
    )

```

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -60,7 +76,7 @@ def _get_supported_currencies(self):
         return supported_currencies
 
-    def _razorpay_make_request(self, endpoint, payload=None, method='POST'):
+    def _razorpay_make_request(self, endpoint, payload=None, method='POST', **kwargs):
```
```suggestion
    def _razorpay_make_request(self, endpoint, payload=None, method='POST', api_version='v1'):
```

It should be like this as there is not point of variable kwargs

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -73,12 +89,11 @@ def _razorpay_make_request(self, endpoint, payload=None, method='POST'):
-
-        # TODO: Make api_version a kwarg in master.
-        api_version = self.env.context.get('razorpay_api_version', 'v1')
+        api_version = kwargs.get('razorpay_api_version', 'v1')
```
adept according to above comment

kcv-odoo commented on addons/payment_razorpay/views/payment_provider_views.xml:
```diff
@@ -6,22 +6,72 @@
                        invisible="code != 'razorpay'">
+                    <label for="razorpay_account_id"/>
+                    <div class="o_row">
+                        <field name="razorpay_account_id"
+                            groups="base.group_no_one"/>
```
```suggestion
                        <field
                            name="razorpay_account_id"
                            groups="base.group_no_one"
                        />
```

kcv-odoo commented on addons/payment_razorpay/views/payment_provider_views.xml:
```diff
@@ -6,22 +6,72 @@
+                                readonly="True"
+                            />
+                    </div>
                     <field name="razorpay_key_id"
```
Can adept this also same as above

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -25,18 +31,28 @@ class PaymentProvider(models.Model):
         string="Razorpay Key Id",
         help="The key solely used to identify the account with Razorpay.",
-        required_if_provider='razorpay',
+        required_if_provider=False,
```
Do we need this? you can just remove required_if_provider param. No? :thinking:

kcv-odoo reviewed (changes requested):
Commit title should be CLN not IMP and added some minor comments!

Thank you for you work!

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -4,13 +4,18 @@
 import logging
 import pprint
-
+import uuid
 import requests
```
Swap this

kcv-odoo commented on addons/payment_razorpay/models/payment_provider.py:
```diff
@@ -4,13 +4,18 @@
 import requests
 
+from urllib.parse import urlencode
+from datetime import timedelta
```
same

kcv-odoo reviewed (approved):
Look good to me (just a minor comment)

AntoineVDV reviewed (approved):
Hi @nikj-odoo, thanks for this PR :)

I pushed an additional commit with further cleanup (some leftovers that weren't properly moved from `payment_razorpay_oauth` and some other stuff). Could you have a look and squash the commits?

Also, please move forward with the upgrade PR so we can r+ both. Note that I didn't rebase this branch so you shouldn't need to either.

nikj-odoo commented:
Hello @AntoineVDV  :wave: : 
I squashed the commit and updated the upgrade [PR](https://github.com/odoo/upgrade/pull/6858), please have a look.

Thanks! & Have a nice day :slightly_smiling_face:

AntoineVDV commented:
@robodoo r+ override=ci/security
(moved with no change (and still okay))

robodoo commented:
@AntoineVDV you may want to rebuild or fix this PR as it has failed CI.
